Sources/Internal/DoubleLayoutPassHelpers.swift

```diff
@@ -0,0 +1,136 @@
+/// information to return an accurate `intrinsicContentSize`. None of this is terribly surprising, since the documentation for
+/// `intrinsicContentSize` specifically says "...intrinsic size must be independent of the content frame,
+/// because there’s no way to dynamically communicate a changed width to the layout system based on a changed height, for
+/// example."
+///
+/// There is one exception to this rule: `UILabel`. When `-[UILabel setNumberOfLines:]` is called with `0` as the number of
+/// lines, the private method `-[UIView _needsDoubleUpdateConstraintPass]` will return `true`. UIKit will then invoke
+/// `-[UILabel intrinsicContentSize]` multiple times with the most up-to-date `preferredMaxLayoutWidth`, enabling
+/// the label to size correctly.
+///
+/// `CalendarView` can leverage `UILabel`'s double layout pass behavior by calling into this extension. Under the hood, a sizing
+/// `UILabel` is used to get a second layout pass, giving `CalendarView` a chance to size itself knowing its final width.
+
+extension CalendarView {
+
+  func installDoubleLayoutPassSizingLabel() {
+    doubleLayoutPassSizingLabel.removeFromSuperview()
+    addSubview(doubleLayoutPassSizingLabel)
+    subviews.first.map(doubleLayoutPassSizingLabel.sendSubviewToBack(_:))
+
+    doubleLayoutPassSizingLabel.translatesAutoresizingMaskIntoConstraints = false
+    NSLayoutConstraint.activate([
+      doubleLayoutPassSizingLabel.leadingAnchor.constraint(
+        equalTo: layoutMarginsGuide.leadingAnchor),
+      doubleLayoutPassSizingLabel.trailingAnchor.constraint(
+        equalTo: layoutMarginsGuide.trailingAnchor),
+      doubleLayoutPassSizingLabel.topAnchor.constraint(equalTo: layoutMarginsGuide.topAnchor),
+      doubleLayoutPassSizingLabel.bottomAnchor.constraint(equalTo: layoutMarginsGuide.bottomAnchor),
+    ])
+  }
+
+  public override func invalidateIntrinsicContentSize() {
+    doubleLayoutPassSizingLabel.invalidateIntrinsicContentSize()
+  }
+
+  public override func setContentHuggingPriority(
+    _ priority: UILayoutPriority,
+    for axis: NSLayoutConstraint.Axis)
+  {
+    doubleLayoutPassSizingLabel.setContentHuggingPriority(priority, for: axis)
+  }
+
+  public override func setContentCompressionResistancePriority(
+    _ priority: UILayoutPriority,
+    for axis: NSLayoutConstraint.Axis)
+  {
+    doubleLayoutPassSizingLabel.setContentCompressionResistancePriority(priority, for: axis)
+  }
+
+}
+
+// MARK: - WidthDependentIntrinsicContentHeightProviding
+
+protocol WidthDependentIntrinsicContentHeightProviding: CalendarView {
+
+  func intrinsicContentSize(forHorizontallyInsetWidth width: CGFloat) -> CGSize
+
+}
+
+// MARK: - DoubleLayoutPassSizingLabel
+
+final class DoubleLayoutPassSizingLabel: UILabel {
+
+  // MARK: Lifecycle
+
+  init(provider: WidthDependentIntrinsicContentHeightProviding) {
+    self.provider = provider
+
+    super.init(frame: .zero)
+
+    numberOfLines = 0
+    isUserInteractionEnabled = false
```

Comment:
I know this is the default but just being I'd prefer to be explicit when doing crazy stuff lol